Lib/test/test_locale.py

```diff
@@ -533,6 +533,12 @@ def test_defaults_UTF8(self):
             if orig_getlocale is not None:
                 _locale._getdefaultlocale = orig_getlocale
 
+    def test_getencoding(self):
+        # Invoke getencoding to make sure it does not cause exceptions.
+        enc = locale.getencoding()
+        # make sure it is valid
+        codecs.lookup(enc)
```

Comment:
_PyUnicode_InitEncodings() fails it config.filesystem_encoding or config.stdio_encoding is not known by codecs.lookup(name). So this call should not fail.

If tomorrow this test fails, I suggest to remove it and only check that the string is non-empty.

Reply:
When UTF-8 mode is enabled, both of stdio encoding and filesystem encoding are UTF-8, not locale encoding.